app/src/main/java/com/lightricks/feedexercise/network/DataTransferObjects.kt

```diff
@@ -1,5 +1,22 @@
 package com.lightricks.feedexercise.network
 
-/**
- * todo: add Data Transfer Object data class(es) here
- */
+import com.squareup.moshi.Json
+import com.squareup.moshi.JsonClass
+
+@JsonClass(generateAdapter = true)
+data class FeedData(@Json(name = "templatesMetadata") val metadata: List<Metadata> = listOf()) {
+    @JsonClass(generateAdapter = true)
+    data class Metadata(
```

Comment:
A bit agreed, mostly disagree: In most cases, the body of an item is an integral part of the request, of your "contract" with the server. Different requests may have other fields. Metadata should represent only the API with the server. Generalisation should be done inside the app structures, not in the server API structures, or it will cripple your ability to encapsulate.

Reply:
How about a request that fetches a single Metadata item? Do you suggest to create a separate data class or to reuse this one?